src/openms/source/ANALYSIS/DENOVO/CompNovoIdentification.cpp

```diff
@@ -897,43 +901,44 @@ namespace OpenMS
     // for each possible peptide charge state
     std::map<Size, double> correlation_sums;
     std::map<Size, std::map<Size, pair<double, double> > > best_corr_ints;
-    for (std::map<Size, std::map<Size, vector<double> > >::const_iterator it1 = correlations.begin(); it1 != correlations.end(); ++it1)
+    for (const auto& corr : correlations)
     {
       double correlation_sum(0);
       // search for the best correlation
-      for (std::map<Size, vector<double> >::const_iterator it2 = it1->second.begin(); it2 != it1->second.end(); ++it2)
+      for (const auto& precs : corr.second)
       {
         double best_correlation(0);
         Size best_pos(0);
         Size i = 0;
-        for (vector<double>::const_iterator it3 = it2->second.begin(); it3 != it2->second.end(); ++it3, ++i)
+        for (const auto& mass : precs.second)
         {
-          if (best_correlation == 0 || *it3 > (best_correlation * 1.25))           // must be really better!
+          if (best_correlation == 0 || mass > (best_correlation * 1.25))           // must be really better!
           {
-            best_correlation = *it3;
+            best_correlation = mass;
             best_pos = i;
           }
+          ++i;
         }
-        best_corr_ints[it1->first][it2->first] = make_pair(best_correlation, peaks[it1->first][it2->first][best_pos].getMZ());
+        best_corr_ints[corr.first][precs.first] = make_pair(best_correlation, peaks[corr.first][precs.first][best_pos].getMZ());
         correlation_sum += best_correlation;
       }
 
-      correlation_sums[it1->first] = correlation_sum;
+      correlation_sums[corr.first] = correlation_sum;
     }
 
     double best_correlation = 0;
     Size best_charge = 0;
-    for (std::map<Size, double>::const_iterator it = correlation_sums.begin(); it != correlation_sums.end(); ++it)
+    for (const auto& corr_sum : correlation_sums)
```

Comment:
e.g., here auto is detrimental for readability and a structured binding would be better